cmd/common/config.go

```diff
@@ -155,7 +155,6 @@ func NewConfigCmd(runtime runtime.Runtime, rootFlags *pflag.FlagSet) *cobra.Comm
 			opGlobalParams.CopyToMap(operatorConfig[opName], "")
 		}
 
-		operatorConfig[ocihandler.OciHandler.Name()] = make(map[string]string)
```

Comment:
It seems we still need something like:

```
		if operatorConfig[ocihandler.OciHandler.Name()] == nil {
			operatorConfig[ocihandler.OciHandler.Name()] = make(map[string]string)
		}
```

otherwise we can have following issues:

```
→ go run ./cmd/kubectl-gadget/ config default
INFO[0000] Experimental features enabled                
WARN[0000] version skew detected: client (v0.0.0) vs server (v0.30.0-160-g2af39a9a) 
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/inspektor-gadget/inspektor-gadget/pkg/params.(*Params).CopyToMap(0xc0007028c0?, 0x0, {0x0, 0x0})
	/home/qasim/work/github.com/inspektor-gadget/inspektor-gadget/pkg/params/params.go:285 +0x17f
github.com/inspektor-gadget/inspektor-gadget/cmd/common.NewConfigCmd.func1(0xc0009dea00?, {0x657e560?, 0x4?, 0x3e4b208?})
	/home/qasim/work/github.com/inspektor-gadget/inspektor-gadget/cmd/common/config.go:159 +0x8d0
github.com/spf13/cobra.(*Command).execute(0xc0009d5208, {0x657e560, 0x0, 0x0})
	/home/qasim/.gvm/pkgsets/go1.21.2/global/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:985 +0xaca
github.com/spf13/cobra.(*Command).ExecuteC(0x5b68f60)
	/home/qasim/.gvm/pkgsets/go1.21.2/global/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/home/qasim/.gvm/pkgsets/go1.21.2/global/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041
main.main()
	/home/qasim/work/github.com/inspektor-gadget/inspektor-gadget/cmd/kubectl-gadget/main.go:168 +0xc65
exit status 2

```

Reply:
> Why is there differences between the output of kubectl-gadget config default and ig config default.

We only show the params which the client (`kubectl-gadget` or `ig` or `gadgetctl`) are allowed to configure/modify.

> Moreover, in this case, we are totally missing operator.oci.public-key which has a default value.

That is correct since if we were showing `operator.oci.public-key` in the default config users will think they can modify it . Does that makes sense?